client/asset/btc/walletclient.go

```diff
@@ -5,48 +5,62 @@ package btc
 
 import (
 	"bytes"
+	"context"
 	"encoding/hex"
 	"encoding/json"
 	"fmt"
 
 	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcd/btcjson"
 	"github.com/btcsuite/btcd/chaincfg"
 	"github.com/btcsuite/btcd/chaincfg/chainhash"
 	"github.com/btcsuite/btcd/wire"
 	"github.com/btcsuite/btcutil"
 )
 
 const (
-	methodGetBalances       = "getbalances"
-	methodListUnspent       = "listunspent"
-	methodLockUnspent       = "lockunspent"
-	methodListLockUnspent   = "listlockunspent"
-	methodChangeAddress     = "getrawchangeaddress"
-	methodNewAddress        = "getnewaddress"
-	methodSignTx            = "signrawtransactionwithwallet"
-	methodUnlock            = "walletpassphrase"
-	methodLock              = "walletlock"
-	methodPrivKeyForAddress = "dumpprivkey"
-	methodSignMessage       = "signmessagewithprivkey"
-	methodGetTransaction    = "gettransaction"
-	methodSendToAddress     = "sendtoaddress"
-	methodSetTxFee          = "settxfee"
-	methodGetWalletInfo     = "getwalletinfo"
-	methodGetAddressInfo    = "getaddressinfo"
+	methodGetBalances        = "getbalances"
+	methodListUnspent        = "listunspent"
+	methodLockUnspent        = "lockunspent"
+	methodListLockUnspent    = "listlockunspent"
+	methodChangeAddress      = "getrawchangeaddress"
+	methodNewAddress         = "getnewaddress"
+	methodSignTx             = "signrawtransactionwithwallet"
+	methodUnlock             = "walletpassphrase"
+	methodLock               = "walletlock"
+	methodPrivKeyForAddress  = "dumpprivkey"
+	methodSignMessage        = "signmessagewithprivkey"
+	methodGetTransaction     = "gettransaction"
+	methodSendToAddress      = "sendtoaddress"
+	methodSetTxFee           = "settxfee"
+	methodGetWalletInfo      = "getwalletinfo"
+	methodGetAddressInfo     = "getaddressinfo"
+	methodEstimateSmartFee   = "estimatesmartfee"
+	methodSendRawTransaction = "sendrawtransaction"
+	methodGetTxOut           = "gettxout"
+	methodGetBlockHash       = "getblockhash"
+	methodGetBestBlockHash   = "getbestblockhash"
+	methodGetRawMempool      = "getrawmempool"
+	methodGetRawTransaction  = "getrawtransaction"
 )
 
+type RawRequester interface {
+	RawRequest(context.Context, string, []json.RawMessage) (json.RawMessage, error)
+}
+
 // walletClient is a bitcoind wallet RPC client that uses rpcclient.Client's
 // RawRequest for wallet-related calls.
 type walletClient struct {
```

Comment:
Should rename this now, since it's far more than just the wallet functions. Now that we don't need the interface, maybe recycle `rpcClient`?